tests/ui/parallel-rustc/deadlock-issue-119785.rs

```diff
@@ -0,0 +1,74 @@
+//@ parallel-front-end
```

Comment:
Discussion: do these tests deserve more context and a basic test doc comment linking to the issue?

Reply:
Yea I add some context for all the cases, but we are still unclear about the causes of some issues (and they may never be reproduced), so they may need to be improved and supplemented in the future.